### PR TITLE
fix set_bipolar_reference for Epochs

### LINKED ISSUE #8726: set_bipolar_reference does not work for epochs data
Somewhat related to my struggles with this function in #8718 . Here is a script to reproduce on master:

```py
import os
import mne

n_picks = 10

sample_data_folder = mne.datasets.sample.data_path()
sample_data_raw_file = os.path.join(sample_data_folder, 'MEG', 'sample',
                                    'sample_audvis_raw.fif')
raw = mne.io.read_raw_fif(sample_data_raw_file, verbose=False).crop(tmax=60)

events = mne.find_events(raw, stim_channel='STI 014')

epochs = mne.Epochs(raw, events, tmin=-0.3, tmax=0.7, preload=True)

epochs.pick_types(meg=False, eeg=True)
ch_names = epochs.info['ch_names']
epochs.pick_channels(ch_names[:n_picks])
ch_names = epochs.ch_names

anode, cathode = ch_names[1:], ch_names[:-1]

evoked = epochs.average()
evoked_bipolar1 = mne.set_bipolar_reference(evoked, anode=anode, cathode=cathode)

epochs_bipolar = mne.set_bipolar_reference(epochs, anode=anode, cathode=cathode)
evoked_bipolar2 = epochs_bipolar.average()
```

I get:

```py
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
~/Documents/github_repos/mne-python/tutorials/epochs/plot_10_epochs_overview.py in <module>
     23 evoked_bipolar1 = mne.set_bipolar_reference(evoked, anode=anode, cathode=cathode)
     24 
---> 25 epochs_bipolar = mne.set_bipolar_reference(epochs, anode=anode, cathode=cathode)
     26 evoked_bipolar2 = epochs_bipolar.average()
     27 

<decorator-gen-232> in set_bipolar_reference(inst, anode, cathode, ch_name, ch_info, drop_refs, copy, verbose)

~/Documents/github_repos/mne-python/mne/io/reference.py in set_bipolar_reference(inst, anode, cathode, ch_name, ch_info, drop_refs, copy, verbose)
    504     if drop_refs:
    505         drop_channels = list((set(anode) | set(cathode)) & set(inst.ch_names))
--> 506         inst.drop_channels(drop_channels)
    507 
    508     return inst

~/Documents/github_repos/mne-python/mne/channels/channels.py in drop_channels(self, ch_names)
    894                    if ch in self.ch_names]
    895         idx = np.setdiff1d(np.arange(len(self.ch_names)), bad_idx)
--> 896         return self._pick_drop_channels(idx)
    897 
    898     def _pick_drop_channels(self, idx):

~/Documents/github_repos/mne-python/mne/channels/channels.py in _pick_drop_channels(self, idx)
    909 
    910         if getattr(self, 'picks', None) is not None:
--> 911             self.picks = self.picks[idx]
    912 
    913         if getattr(self, '_read_picks', None) is not None:

IndexError: index 10 is out of bounds for axis 0 with size 10
```

### PR DESCRIPTION
**Pull request by agramfort**
fix #8726

### CODE REVIEW

**Comment by agramfort** [comment]
please have a look @jasmainak

File: doc/changes/latest.inc [hunk 1]
```diff
@@ -35,6 +35,8 @@ Bugs
 
 - Fix bug with :func:`mne.io.read_raw_nicolet` where header type values such as num_sample and duration_in_sec where not parsed properly (:gh:`8712` by `Alex Gramfort`_)
 
+- Fix bug with :func:`mne.set_bipolar_reference` when passing Epochs (:gh:`8728` by `Alex Gramfort`_)
```
Comment (jasmainak) [inline doc/changes/latest.inc:38]: maybe a bit more explicit about what the bug was?

File: mne/io/reference.py [hunk 2]
```diff
@@ -500,6 +500,9 @@ def set_bipolar_reference(inst, anode, cathode, ch_name=None, ch_info=None,
         logger.info('Bipolar channel added as "%s".' % name)
         inst.info._update_redundant()
 
+    if getattr(inst, 'picks', None) is not None:
+        del inst.picks  # picks cannot be tracked anymore
```
Comment (jasmainak) [inline mne/io/reference.py:504]: this is new for me, not sure what this does

also not [found in the Attributes section of the docstring](https://mne.tools/dev/generated/mne.Epochs.html#mne.Epochs)

**Review by jasmainak — APPROVED** [review]
Other than minor nitpicks, +1 for merge

File: mne/io/reference.py (hunk 2 above)
Comment (agramfort) [reply mne/io/reference.py:504]: the .picks attribute when you passed picks to the Epochs constructor.

**Comment by jasmainak** [comment]
Thanks a ton @agramfort !

**Pull request merged**